File: Ruby/Sorts/BubbleSort.rb
# Brian Taylor Vann
# https://github.com/taylor-vann

def bubble_sort(arr)
  return arr if arr.length < 2

  swpd = true

  while swpd
    swpd = false

    for j in 0..(arr.length - 2)
      if arr[j] > arr[j + 1]
        arr[j], arr[j + 1] = ar[j + 1], arr[j];
        swpd = true
      end
    end
  end

  return arr
end

File: Java/Sorts/BubbleSort.java
// Brian Taylor Vann
// github.com/taylor-vann

import java.util.Arrays;

class BubbleSort {
  void bubbleSort(int[] arr) {
    if(arr.length < 2) {
      return arr;
    }

    boolean swpd = true;

    // until list is traversed without swapping ...
    while(swpd) {
      swpd = false
      // compare one element to the next for the list ...
      for(int j = 0; j < arr.length - 2; j++) {
        // if prev is less than next then swap ...
        if(arr[j] > arr[j + 1]) {
          int tmp = arr[j];
          arr[j] = arr[j + 1];
          arr[j + 1] = tmp;

          swpd = true;
        }
      }
    }
  }


  public static void main(String[] args) {
    BubbleSort bub = new BubbleSort();

    int[] arr = {1, 67, 4, 8, 3, 7};
    bub.bubbleSort(arr);

    System.out.println(Arrays.toString(arr));
  }
}
